exporter/signalfxexporter/exporter_test.go

```diff
@@ -167,3 +172,196 @@ func generateLargeBatch(t *testing.T) *consumerdata.MetricsData {
+								"property2": "val2",
+							},
+							MetadataToUpdate: map[string]string{
+								"prop.erty3": "val33",
+								"property4":  "",
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			"Test tag updates",
+			fields{
+				map[string]interface{}{
+					"customProperties": map[string]interface{}{},
+					"tags": []interface{}{
+						"tag_1",
+					},
+					"tagsToRemove": []interface{}{
+						"tag_2",
+					},
+				},
+			},
+			args{
+				[]*collection.KubernetesMetadataUpdate{
+					{
+						ResourceIDKey: "key",
+						ResourceID:    "id",
+						MetadataDelta: collection.MetadataDelta{
+							MetadataToAdd: map[string]string{
+								"tag.1": "",
+							},
+							MetadataToRemove: map[string]string{
+								"tag/2": "",
+							},
+							MetadataToUpdate: map[string]string{},
+						},
+					},
+				},
+			},
+		},
+		{
+			"Test quick successive updates",
+			fields{
+				map[string]interface{}{
+					"customProperties": map[string]interface{}{
+						"property1": nil,
+						"property2": "val2",
+						"property3": nil,
+					},
+					"tags": []interface{}{
+						"tag_2",
+					},
+					"tagsToRemove": []interface{}{
+						"tag_1",
+					},
+				},
+			},
+			args{
+				[]*collection.KubernetesMetadataUpdate{
+					{
+						ResourceIDKey: "key",
+						ResourceID:    "id",
+						MetadataDelta: collection.MetadataDelta{
+							MetadataToAdd: map[string]string{
+								"tag.1":     "",
+								"property1": "val1",
+								"property3": "val3",
+							},
+							MetadataToRemove: map[string]string{
+								"tag/2": "",
+							},
+							MetadataToUpdate: map[string]string{
+								"property2": "val22",
+							},
+						},
+					},
+					{
+						ResourceIDKey: "key",
+						ResourceID:    "id",
+						MetadataDelta: collection.MetadataDelta{
+							MetadataToAdd: map[string]string{
+								"tag/2": "",
+							},
+							MetadataToRemove: map[string]string{
+								"tag.1":     "",
+								"property1": "val1",
+							},
+							MetadataToUpdate: map[string]string{
+								"property2": "val2",
+								"property3": "val33",
+							},
+						},
+					},
+					{
+						ResourceIDKey: "key",
+						ResourceID:    "id",
+						MetadataDelta: collection.MetadataDelta{
+							MetadataToAdd: map[string]string{},
+							MetadataToRemove: map[string]string{
+								"property3": "val33",
+							},
+							MetadataToUpdate: map[string]string{},
+						},
+					},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				b, err := ioutil.ReadAll(r.Body)
+				require.NoError(t, err)
+
+				p := map[string]interface{}{
+					"customProperties": map[string]*string{},
+					"tags":             []string{},
+					"tagsToRemove":     []string{},
+				}
+
+				err = json.Unmarshal(b, &p)
+				require.NoError(t, err)
+
+				require.Equal(t, tt.fields.payLoad, p)
```

Comment:
it's not guaranteed that this check is executed, right? can we add a synchronization to make sure it's executed for every subtest?

Reply:
That's a good point. Thanks for catching it. Will update.